Gemfile

```diff
@@ -8,6 +8,9 @@ gem "rails", "~> 7.0.8"
 
 # Use postgresql as the database for Active Record
 gem "pg", "~> 1.1"
+gem "rom"
+gem "rom-sql"
+gem "dry-validation"
```

Comment:
One of the requirements was to be able to pass database configuration into the `Geolocation` component. I couldn't find that `ActiveRecord` allows doing so, that's why I decided to use `rom-rb`.
Also, `rom-rb` is similar to `ecto`, it brings repositories, schemas, etc.